src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -1395,6 +1398,42 @@ public void storeDOIPreferences(DOIPreferences preferences) {
         put(BASE_DOI_URI, preferences.getDefaultBaseURI());
     }
 
+    /**
+     * Gets the CustomAPIKeyPreferences that contains information: the custom API key and whether to use it
+     *
+     * @param name the key name of API key
+     * @return CustomAPIKeyPreferences
+     */
+    @Override
+    public CustomApiKeyPreferences getCustomApiKeyPreferences(String name) {
+        return new CustomApiKeyPreferences(name,
+                getBoolean(USE_CUSTOM_API_KEY + name, false),
+                get(CUSTOM_API_KEY + name, ""));
+    }
+
+    /**
+     * Saves CustomAPIKeyPreferences information: the custom custom API key and whether to use it
+     *
+     * @param preferences CustomApiKeyPreferences
+     */
+    @Override
+    public void storeCustomApiKeyPreferences(CustomApiKeyPreferences preferences) {
+        String keyName = preferences.getName();
+        putBoolean(USE_CUSTOM_API_KEY + keyName, preferences.isUseCustom());
+        put(CUSTOM_API_KEY + keyName, preferences.getCustomApiKey());
+    }
+
+    /**
+     * Removes the value associated with the specified custom api key in this preference node, if any.
+     *
+     * @param name the name of custom api key
+     */
+    @Override
+    public void clearCustomApiKeyPreferences(String name) {
+        prefs.remove(USE_CUSTOM_API_KEY + name);
+        prefs.remove(CUSTOM_API_KEY + name);
+    }
```

Comment:
I think it's better to store the keys as a list but to make some constant-name-vodoo. Make to lists, one with the names, the second with the keys, maybe the third with the use flag. then create a cache for lazy loading and output the keys for every key when asked.

Reply:
The value stored in Preferences seems to be only int, long, double, float, boolean, bytearray, String.